### PR TITLE
Unset moment.js locale in timezone service

### PR DESCRIPTION
**Pull request by dombesz**
https://community.openproject.org/wp/48834

### CODE REVIEW

**Review by oliverguenther — CHANGES_REQUESTED** [review]
@dombesz Can you elaborate on why this would fix the issue? Removing these lines would prevent localized output of dates in e.g, the work package cards

**Comment by dombesz** [comment]
The error is related to the localization mechanism in moment.js. In some languages, such as Arabic, Bengali, Burmese, Cambodian, Hindi etc, the numbers are also translated from traditional arabic `2024-01-30` to the language's native characters such as `៣០/០១/២០២៤`, `٣٠/‏١/‏٢٠٢٤` etc.
I just removed the localization of the date formatter to see how many tests would be affected by that.

After some analysis, the solution I see has 2 parts:
- Respect the user's language format from the settings `admin/settings/date_format` in the datepicker.
- Translate the dates back to a machine readable date format before sending them to the server.

**Comment by oliverguenther** [comment]
Both your suggested changes would require a big change in the frontend (not saying I'm rejecting either, but would lean towards the second - using the Intl. APIs to access dates instead of moment).

You mentioned we could try to disable the locale setting just for parsing the ISO8601 dates (or use another mechanism to parse these).

File: frontend/src/app/core/datetime/timezone.service.ts [hunk 1]
```diff
@@ -37,13 +37,7 @@ export class TimezoneService {
   constructor(
     readonly configurationService:ConfigurationService,
     readonly I18n:I18nService,
-  ) {
-    this.setupLocale();
-  }
-
-  public setupLocale():void {
```
Comment (dombesz) [inline frontend/src/app/core/datetime/timezone.service.ts:44]: ℹ️This is no more required, because we set the locale in the `init-locale.ts` below.

**Review by oliverguenther — APPROVED** [review]

**Pull request merged**